onnxruntime/core/mlas/lib/mlasi.h

```diff
@@ -210,9 +229,19 @@ class MLASCPUIDInfo
     bool has_arm_sve_{false};
     bool has_arm_sve_i8mm_{false};
     bool has_arm_neon_bf16_{false};
+    bool has_arm_sme_{false};
+    bool has_arm_sme2_{false};
 };
 using MLAS_CPUIDINFO = MLASCPUIDInfo;
 
+// Boolean condition to determine if we can use SME2
+// By default we should try for SME2 first before falling back to SME.
+inline const bool SMEInfo::CanUseSME2 = MLAS_CPUIDINFO::GetCPUIDInfo().HasArm_SME2();
```

Comment:
Can we avoid duplicate initializations by just moving the initializations out of the `BUILD_MLAS_NO_ONNXRUNTIME` guarded sections ?